src/consumer/config.rs

```diff
@@ -0,0 +1,140 @@
+                        let file = File::open(path).map_err(|io_err| {
+                            NjError::Other(format!(
+                                "An error ocurred opening file on {}. {:?}",
+                                file_path, io_err,
+                            ))
+                        })?;
+                        let mut reader = BufReader::new(file);
+                        let mut gz_encoder = GzEncoder::new(Vec::new(), Compression::default());
+                        let mut buff: Vec<u8> = Vec::new();
+
+                        reader.read_to_end(&mut buff).map_err(|io_err| {
+                            NjError::Other(format!(
+                                "Failed to read contents of file on {}: {:?}",
+                                file_path, io_err
+                            ))
+                        })?;
+                        gz_encoder.write_all(buff.as_slice()).map_err(|io_err| {
+                            NjError::Other(format!(
+                                "An error ocurred while reading WASM file on {}: {:?}",
+                                file_path, io_err
+                            ))
+                        })?;
+                        let bytes = gz_encoder.finish().map_err(|io_err| {
+                            NjError::Other(format!(
+                                "An error ocurred while encoding WASM into Gzip. {:?}",
+                                io_err
+                            ))
+                        })?;
+
+                        Ok(Some(SmartModuleInvocation {
+                            wasm: SmartModuleInvocationWasm::AdHoc(bytes),
+                            kind,
+                            ..Default::default()
+                        }))
+                    }
+                    (None, Some(name), None) => Ok(Some(SmartModuleInvocation {
+                        wasm: SmartModuleInvocationWasm::Predefined(name),
+                        kind,
+                        ..Default::default()
+                    })),
+                    (None, None, Some(data)) => {
+                        let wasm = base64::decode(data).map_err(|e| {
+                            NjError::Other(format!(
+                "An error ocurred attempting to decode the Base64 WASM file provided. {:?}",
+                e
+            ))
+                        })?;
+
+                        Ok(Some(SmartModuleInvocation {
+                            wasm: SmartModuleInvocationWasm::AdHoc(wasm),
+                            kind,
+                            ..Default::default()
+                        }))
+                    }
+                    _ => Err(NjError::Other(format!(
+                        "You must either provide one of {}, {} or {}",
+                        CONFIG_SMART_MODULE_FILE_KEY,
+                        CONFIG_SMART_MODULE_NAME_KEY,
+                        CONFIG_SMART_MODULE_DATA_KEY
+                    ))),
```

Comment:
If the user provides more than 1 source for the SM, the state will be ambiguous, to avoid this an error is thrown complaining about the input.